### PR TITLE
updated the logging in the notify controller

### PR DESCRIPTION
**Pull request by bkhan2**
# Motivation and Context
updating logging when notify notification doesn't get sent to be clearer

# What has changed
added URL and Status Code logging.

### CODE REVIEW

File: ras_party/controllers/notify_gateway.py [hunk 1]
```diff
@@ -35,25 +35,25 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
...
+
+        url = urlparse.urljoin(self.notify_url, str(template_id))
+        response = Requests.post(url, json=notification)
+        status_code = response.status_code
+
+        if status_code == 201:
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json()["id"])
-
-        except Exception as e:
+        else:
```
Comment (jterryons) [inline ras_party/controllers/notify_gateway.py]: is the deleted exception never thrown? we're only checking here for a 201. but codes 200 - 399 are valid and shouldnt raise an error. i think theres a method that might be useful response.raise_for_status() or something similar?

**Review by jterryons — CHANGES_REQUESTED** [review]

File: ras_party/controllers/notify_gateway.py (hunk 1 above)
Comment (mfcrocker) [reply ras_party/controllers/notify_gateway.py]: I disagree with this - 201 is the only valid non-error response from this endpoint and if we're considering the APIs a contract we should be willing to code that in

File: ras_party/controllers/notify_gateway.py [hunk 2]
```diff
@@ -35,25 +35,25 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
...
+
+        if 201 <= status_code <= 399:
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json()["id"])
-
-        except Exception as e:
+        else:
             ref = reference if reference else 'reference_unknown'
-            raise RasNotifyError("There was a problem sending a notification to Notify-Gateway to GOV.UK Notify",
-                                 error=e, reference=ref)
+            raise RasNotifyError("There was a problem sending a notification to Notify-Gateway to GOV.UK Notify."
```
Comment (mfcrocker) [inline ras_party/controllers/notify_gateway.py]: You don't need to pass these variables in as fstrings, `RasNotifyError` takes kwargs
```suggestion
            raise RasNotifyError("There was a problem sending a notification to Notify-Gateway.", url=url, status_code=status_code, message=response.text, reference=ref)
```

**Review by mfcrocker — CHANGES_REQUESTED** [review]
Same comment around fstrings

File: ras_party/controllers/notify_gateway.py (hunk 1 above)
Comment (jterryons) [reply ras_party/controllers/notify_gateway.py]: a 200/300 status code here will still have created the resource in whatever service is being called and will presumably be in a valid state. this error will not roll back any successful transactions and will be a nightmare to investigate any ops issue as a result of the error thrown. If a 201 is the only valid response then there should be no need to check it and a raise_for_status should suffice.

**Comment by jterryons** [comment]
Similar to the frontstage change, this isnt worth the test effort.

**Pull request closed without merging**